### ISSUE TITLE
[HOLD for payment 2023-06-08] [$1000] Error appears for a second when user select new member as Assignee

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19203]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. click + button
2. select Assign task option
3. Enter tittle and click on Next button
4. click on Assignee option
5. search for new member which is not in list
6. select that member

## Expected Result: 
Error appears for a second on Confirm task page
## Actual Result: 
should not show any error

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.16-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/39495e4c-2380-4257-b5e5-2fb67a68068d

https://github.com/Expensify/App/assets/43996225/a6fec959-8f9a-4eea-b2a1-8f74fac29987

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684243797708189

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0112661f5cf190cf47</li>
        <li>Upwork Job ID: 1660712581755756544</li>
        <li>Last Price Increase: 2023-05-22</li>
    </ul>
</details>

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Error appears for a second when user select new member as Assignee
### What is the root cause of that problem?
If the assignee is new member, `props.personalDetails` will not have detail information of that member until API openReport that is called in `setAssigneeValue` returns data
https://github.com/Expensify/App/blob/9111b90468cd92867b93bf823e37060af819686a/src/libs/actions/Task.js#L429-L445
So that before API openReport returns data,  `assigneeDetails` is undefined and the error appears. Thus issue occours
https://github.com/Expensify/App/blob/9111b90468cd92867b93bf823e37060af819686a/src/pages/tasks/NewTaskPage.js#L78-L86
### What changes do you think we should make in order to solve the problem?
If expected we can assign for new member that doesn't exist in `personalDetails` before, we could get `assigneeDetails` from `OptionsListUtils.getPersonalDetailsForLogins([assignee], props.personalDetails)` instead of `props.personalDetails`
https://github.com/Expensify/App/blob/9111b90468cd92867b93bf823e37060af819686a/src/pages/tasks/NewTaskPage.js#L79

if it's not expected to create assign task for new member, we should only call openReport in here if assignee is not a new chat
https://github.com/Expensify/App/blob/9111b90468cd92867b93bf823e37060af819686a/src/libs/actions/Task.js#L441
### What alternative solutions did you explore? (Optional)

**Comment by sakluger** [comment]
I was able to reproduce this error. I also discussed in Slack with @thienlnam here (https://expensify.slack.com/archives/C04QEB4MJEQ/p1684539286006029). I think that this is happening when you assign a task to someone who isn't in the current DM/room you are in.

Asked Jack if this should be internal or external, just waiting for that answer before moving this forward.

**Comment by thienlnam** [comment]
Adding external so we can get some investigation on how to moveforward

**Comment by sobitneupane** [comment]
Thanks for the proposal @dukenv0307.

- This part of the code was probably introduced to get rid of https://github.com/Expensify/App/pull/17992#issuecomment-1535567020 issue. 

https://github.com/Expensify/App/blob/9111b90468cd92867b93bf823e37060af819686a/src/pages/tasks/NewTaskPage.js#L79-L83

- Even though error is displayed, user are allowed to "Confirm task" which should not be the case.

- User won't be able to assign task to new user in offline if user has to wait for api response for personal details.

Based on the above points, I don't think we want to block to assign task to a new user.

[Proposal](https://github.com/Expensify/App/issues/19203#issuecomment-1554043255) from @dukenv0307 to use `OptionsListUtils.getPersonalDetailsForLogins` looks good.

🎀👀🎀 C+ reviewed
cc: @jasperhuangg

**Comment by jasperhuangg** [comment]
agree! let's go with their proposal, thanks!

**Comment by dukenv0307** [comment]
@jasperhuangg  @sobitneupane  The PR is ready to review

**Comment by sakluger** [comment]
Now that we've hired everyone, I've sent contracts to everyone on Upwork.

**Comment by mallenexpensify** [comment]
- https://github.com/Expensify/App/issues/19462#issuecomment-1564988403

Coming from the above, do we think the fix for this issue might fix #19462?  If so I can put that on hold.

**Comment by sakluger** [comment]
Paid for the bug report. I'll be OOO on 6/8 when we need to pay everyone else, so I'm going to assign another BZ team member to complete payment.

@sobitneupane can you please make sure to complete the BugZero checklist between now and then? Thanks!

**Comment by tjferriss** [comment]
On track to pay in a couple days. @sobitneupane can you help with the checklist.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/17992
> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/17992#issuecomment-1584086592
> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

I don't think any change in checklist is required. It should have been caught in offline test.

> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug. 

yes

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

1. Go to FAB > Assign Task > Enter title and description and navigate to next page
2. Click on Assignee and enter a new account which is not used anywhere yet
3. Verify that the assignee is added immediately and no error appears

Do we agree 👍 or 👎

**Comment by tjferriss** [comment]
Payments have been sent including the 50% speed bonus. I've created the regression test issue for Applause.
